### PR TITLE
Select rules based on reference

### PR DESCRIPTION
**Pull request by jan-cerny**
This PR introduces a new option `--reference` that allows users to filter rules base on rule reference. This is useful to evaluate compliance with a particular policy requirement.

This feature needs SCAP content to contain reference URI-to-title mapping within the `xccdf:Benchmark` element. The ComplianceAsCode content upstream project started to provide this mapping in the SCAP content recently, starting from 62513c391dc5a3fafd12741bd02565ca0e1e8db2.

To support the easier experience with this feature, also a new option `--references` to the `oscap info` module has been added, this option extends the output of the command by listing all available reference names.

Fixes: https://issues.redhat.com/browse/RHEL-1479

### CODE REVIEW

File: utils/oscap-xccdf.c [hunk 1]
```diff
@@ -153,6 +153,7 @@ static struct oscap_module XCCDF_EVAL = {
 		"   --profile <name>              - The name of Profile to be evaluated.\n"
 		"   --rule <name>                 - The name of a single rule to be evaluated.\n"
 		"   --skip-rule <name>            - The name of the rule to be skipped.\n"
+		"   --reference <URL=ID>          - Evaluate only rules that have the given reference.\n"
```
Comment (Mab879) [inline utils/oscap-xccdf.c:156]: Should this a `:`?

File: src/XCCDF_POLICY/xccdf_policy.c [hunk 2]
```diff
@@ -2253,6 +2281,44 @@ void xccdf_policy_model_free(struct xccdf_policy_model * model) {
...
+	return uri;
+}
+
+void xccdf_policy_set_reference_filter(struct xccdf_policy *policy, const char *reference_parameter)
+{
+	if (!reference_parameter) {
+		return;
+	}
+	char *reference_parameter_dup = strdup(reference_parameter);
+	char **split = oscap_split(reference_parameter_dup, ":");
```
Comment (Mab879) [inline src/XCCDF_POLICY/xccdf_policy.c:2306]: So I can get a segfault if I run `./oscap_wrapper xccdf eval --profile stig --reference 'stigid'  [datastream]`. There should be some error handling for this case.

**Review by Mab879 — CHANGES_REQUESTED** [review]
Thanks for the PR.

Please take a look at my comments.

**Comment by jan-cerny** [comment]
I have fixed the segfault. I have fixed the help text.

**Review by Mab879 — APPROVED** [review]
LGTM.

Thanks.

**Review by Mab879 — CHANGES_REQUESTED** [review]
Can you please review the testing farm failure?

**Comment by jan-cerny** [comment]
It's the  probe_behavior/collect_limit.sh which fails randomly from time to time

**Pull request merged**